app/analytics/collect.ts

```diff
@@ -130,7 +130,7 @@ interface DataPoint {
 // More here: https://developers.cloudflare.com/analytics/analytics-engine/get-started/#limits
 
 export function writeDataPoint(
-    analyticsEngine: CFAnalyticsEngine,
+    analyticsEngine: AnalyticsEngineDataset,
```

Comment:
I think generally speaking, this change isn't really related to migrating to `vite`, and more of a separate personal choice that's been inserted (same with renaming `Environment` to `Env`).

I'll keep it (it's more clear I suppose) but generally it's best to not to couple these changes.

Reply:
Agreed, this `AnalyticsEngineDataset` type is directly from `@cloudflare/workers-types`
So, syntaxes and change tracking are aligned with CF.

Regarding `Env`, I personally prefer `Environment` whatever you have set but cloudflare auto generates types from `wrangler.toml` to `worker-configuration.d.ts` by using `wrangler types`, that is the I have removed `Environment` and reused `Env` from `worker-configuration.d.ts`